golang/dbos-toolbox/main.go

```diff
@@ -0,0 +1,113 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/dbos-inc/dbos-transact-go/dbos"
+)
+
+/*****************************/
+/**** WORKFLOWS AND STEPS
+/*****************************/
+var (
+	wf = dbos.WithWorkflow(workflow)
+)
+
+func workflow(ctx context.Context, _ string) (string, error) {
+	_, err := dbos.RunAsStep(ctx, step1, "")
+	if err != nil {
+		return "", err
+	}
+	return dbos.RunAsStep(ctx, step2, "")
+}
+
+func step1(ctx context.Context, _ string) (string, error) {
+	fmt.Println("Executing step 1")
+	return "Step 1 completed", nil
+}
+func step2(ctx context.Context, _ string) (string, error) {
+	fmt.Println("Executing step 2")
+	return "Step 2 completed - Workflow finished successfully", nil
+}
+
+/*****************************/
+/**** QUEUES
+/*****************************/
+var (
+	queue   = dbos.NewWorkflowQueue("example-queue")
+	step    = dbos.WithWorkflow(stepFunction)
```

Comment:
Should this be `WithStep`?

Reply:
step is a misnomer, we don't have temp workflows and WithStep. Each task of a workflow is a child workflow